akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -233,6 +233,24 @@ private[http] object StreamUtils {
       def open(): Unit = promise.success(())
     }
   }
+
+  /**
+   * Returns a flow that is almost identity but doesn't pass cancellation from downstream to upstream.
+   *
+   * Note: This might create a resource leak if the upstream never completes. External measures
+   * need to be taken to ensure that the upstream will always complete.
+   */
+  def ignoreCancellation[T]: Flow[T, T, NotUsed] = Flow.fromGraph(new IgnoreCancellation)
+  class IgnoreCancellation[T] extends SimpleLinearGraphStage[T] {
```

Comment:
Also, perhaps changing name slightly would be good? Like `BarrierCancellation` / `BarrierCancellation`?
I've had instances where I needed to barrier off completion too, so that would make it naming wise simple to add such one too once needed again